### PR TITLE
Passing DB password into the server container.

### PR DESCRIPTION
**Pull request by eldarm**
TODO: same for DB container.

$./c.sh
2023/12/05 17:18:55 DoUser()
2023/12/05 17:18:55 Done, 200 OK: {
    "team": 2
}
2023/12/05 17:18:55 doReport()
2023/12/05 17:18:55 Done, 200 OK:
2023/12/05 17:18:55 doMap()
2023/12/05 17:18:55 Done, 200 OK: [
    {
        "Latitude": 35.767708,
        "Longitude": -89.857346,
        "Count": 1
    }
]
2023/12/05 17:18:55 doStats()
2023/12/05 17:18:55 Done, 200 OK: {
    "version": "2.0",
    "id": "F8A7A2F04F4A4273",
    "kittens": 1
}
2023/12/05 17:18:55 doTeams()
2023/12/05 17:18:55 Done, 200 OK: {
    "base": {
        "version": "2.0",
        "id": "F8A7A2F04F4A4273"
    },
    "blue": 0,
    "green": 1
}
2023/12/05 17:18:55 doTopScores()
2023/12/05 17:18:55 Done, 200 OK: {
    "records": [
        {
            "place": 1,
            "title": "La Puch da Vinchi",
            "kitn": 1,
            "is_you": true
        }
    ]
}
2023/12/05 17:18:55 doWriteReferral()
2023/12/05 17:18:55 Done, 200 OK:
2023/12/05 17:18:55 doReadReferral()
2023/12/05 17:18:55 Done, 200 OK: {"refvalue":"abcdef"} 2023/12/05 17:18:55 doGenerateReferral()
2023/12/05 17:18:55 Done, 200 OK: {"refvalue":"FvrvjGRyxQ"}

### CODE REVIEW

File: be/db.go [hunk 1]
```diff
@@ -12,9 +12,18 @@ import (
 )
 
 var (
-	mysqlAddress = flag.String("mysql_address", "server:dev_pass@tcp(cleanupdb:3306)/cleanapp", "MySQL address string")
+	mysqlPassword = flag.String("mysql_password", "secret", "MySQL password.")
+	mysqlHost = flag.String("mysql_host", "cleanupdb", "MySQL host.")
```
Comment (eugene-eko2000) [inline be/db.go:16]: Let's make the default value localhost. cleanappdb isn't a default one, at least for running on a localhost. And

File: be/db.go [hunk 2]
```diff
@@ -12,9 +12,18 @@ import (
...
+	mysqlPort = flag.String("mysql_port", "3306", "MySQL port.")
+	mysqlDb = flag.String("mysql_db", "cleanapp", "MySQL database to use.")
 )
 
+func mysqlAddress() string {
+	db := fmt.Sprintf("server:%s@tcp(%s:%s)/%s", *mysqlPassword, *mysqlHost, *mysqlPort, *mysqlDb)
+	// log.Printf("DB connect string: %q", db) // Troubleshooting only. Exposes password.
```
Comment (eugene-eko2000) [inline be/db.go:23]: Let's remove a commented code.

File: docker/Dockerfile [hunk 3]
```diff
@@ -15,6 +15,7 @@ COPY ./service .
 # USER cleanapper
 
 EXPOSE 8080/tcp
-CMD [ "./service" ]
+# CMD [ "./service", "--mysql_password=${MYSQL_APP_PASSWORD}" ] # TODO: FIX!!!
```
Comment (eugene-eko2000) [inline docker/Dockerfile:18]: Let's remove a commented code and make a comment readable i.e. what is wrong with a current state and how to change it

**Review by eugene-eko2000 — APPROVED** [review]

File: be/db.go (hunk 2 above)
Comment (eldarm) [reply be/db.go:23]: It would be useful to me onthe second phase, but ok, done.

File: be/db.go (hunk 1 above)
Comment (eldarm) [reply be/db.go:16]: Done.

File: docker/Dockerfile (hunk 3 above)
Comment (eldarm) [reply docker/Dockerfile:18]: Right, done.

**Pull request merged**